File: Legacy_Manually/PlagueExternal.dll/LocalDiseaseUpdate.cs
void __fastcall LocalDiseaseUpdate(SPLocalDiseaseData localDisease, SPCountryData country, SPDiseaseData disease)
{
  float v6; // xmm0_4
  __int64 infectedPercent; // r9
  __int64 v8; // r8
  float zombieAwareness; // xmm7_4
  float v10; // xmm1_4
  float v11; // xmm1_4
  double v12; // xmm2_8
  float v13; // xmm1_4
  float v14; // xmm2_4
  float v15; // xmm1_4
  float v16; // xmm0_4
  __int64 v17; // rdx
  float v18; // xmm1_4
  __int64 v19; // rcx
  float v20; // xmm3_4
  float v21; // xmm0_4
  int totalImportance; // eax
  float resistanceRaw; // xmm3_4
  float apeResistance; // xmm2_4
  int coountry.wealthy; // ecx
  float localResistance; // xmm3_4
  float v27; // xmm2_4
  float v28; // xmm0_4
  float v29; // xmm1_4
  float v30; // xmm2_4
  float localSeverity; // xmm1_4
  float localSeveritySpecial; // xmm4_4
  float v33; // xmm5_4
  float localSeverityTrue; // xmm1_4
  int disease.diseaseType; // ebp
  float environmentalSeverity; // xmm1_4
  float SeverityPower; // xmm2_4
  float environmentalSeverityModified; // xmm0_4
  float environmentalSeverityModifiedPlus; // xmm0_4
  float severityRate; // xmm1_4
  float constK; // xmm0_4
  float infectedAwareness; // xmm6_4
  float deadAwareness; // xmm8_4
  float deadAwarenessModified; // xmm0_4
  float dangerAwareness; // xmm6_4
  int difficultyConst; // eax

  if ( *(_DWORD *)(country._borderStatus) )
    v6 = 1.0;
  else
    v6 = 0.050000001;
  *(float *)(localDisease.borderStatusValue) = v6;
  if ( *(_DWORD *)(disease._closedBordersSpreadEnhance) )
    *(float *)(localDisease.borderStatusValue) = v6 * 3.0;
  infectedPercent = *(_QWORD *)(a1 + 16) + *(_QWORD *)(a1 + 32);
  v8 = *(_QWORD *)(a1 + 264);
  zombieAwareness = 0.0;
  v10 = (double)(int)*(_QWORD *)(a1 + 8) / (double)(int)*(_QWORD *)(a2 + 24);
  *(float *)(localDisease.deadPercent) = v10;
  v11 = (double)(int)infectedPercent / (double)(int)*(_QWORD *)(a2 + 24);
  v12 = (double)(int)*(_QWORD *)(a1 + 24);
  *(float *)(localDisease.infectedPercent) = v11;
  v13 = v12 / (double)(int)*(_QWORD *)(a2 + 24);
  v14 = 0.0;
  *(float *)(localDisease.zombiePercent) = v13;
  *(float *)(a1 + 288) = (float)(int)*(_QWORD *)(a2 + 208) / (float)(int)*(_QWORD *)(a2 + 24);
  v15 = (float)(int)*(_QWORD *)(a2 + 200);
  v16 = (float)(int)*(_QWORD *)(a2 + 24);
  v17 = *(_QWORD *)(a1 + 272);
  *(float *)(a1 + 292) = v15 / v16;
  v18 = 0.0;
  v19 = *(_QWORD *)(a1 + 280);
  v20 = (float)(v19 + v8 + v17);
  if ( v20 >= 0.5 )
  {
    v21 = (float)(int)v17 / v20;
    v18 = (float)(int)v8 / v20;
    v14 = (float)(int)v19 / v20;
  }
  else
  {
    v21 = 1.0;
  }
  *(float *)(a1 + 320) = v14;
  *(float *)(a1 + 316) = v18;
  totalImportance = 0;
  *(float *)(a1 + 324) = v21;
  resistanceRaw = 0.0;
  *(_QWORD *)(a1 + 156) = 0i64;
  apeResistance = 1.0;
  *(_QWORD *)(a1 + 148) = 0i64;
  *(_QWORD *)(a1 + 140) = 0i64;
  *(_DWORD *)(a1 + 136) = 0;
  *(_QWORD *)(a1 + 304) = 0i64;
  *(_DWORD *)(a1 + 300) = 0;
  if (disease.diseaseType == SIMIAN_FLU)
  {
    if (coountry.wealthy)
    {
      totalImportance = 6;
      apeResistance = 0.9f;
      resistanceRaw = disease.wealthy * 6.0;
    }
    if (coountry.poverty)
    {
      totalImportance += 3;
      resistanceRaw = resistanceRaw + disease.poverty * 3.0;
      apeResistance = apeResistance * 1.1f;
    }
    if (coountry.urban)
    {
      totalImportance += 2;
      resistanceRaw = resistanceRaw + disease.urban * 2.0;
      apeResistance = apeResistance * 0.8f;
    }
    if (coountry.rural)
    {
      totalImportance += 2;
      resistanceRaw = resistanceRaw + disease.rural * 2.0;
      apeResistance = apeResistance * 1.2f;
    }
    if (coountry.hot)
    {
      totalImportance += 8;
      resistanceRaw = resistanceRaw + disease.hot * 8.0;
      apeResistance = apeResistance * 0.9f;
    }
    if (coountry.cold)
    {
      totalImportance += 10;
      resistanceRaw = resistanceRaw + disease.cold * 10.0;
      apeResistance = apeResistance * 0.9f;
    }
    if (coountry.humid)
    {
      totalImportance += 2;
      resistanceRaw = resistanceRaw + disease.humid * 2.0;
      apeResistance = apeResistance * 1.1f;
    }
    if (coountry.arid)
    {
      resistanceRaw = resistanceRaw + disease.arid * 2.0;
      apeResistance = apeResistance * 0.75f;
      totalImportance += 2;
    }
  }
  else
  {
    if ( coountry.wealthy )
    {
      totalImportance = 30;
      resistanceRaw = (float)(*(float *)(disease.wealthy) * 30.0) + 0.0;
    }
    if ( *(_DWORD *)(coountry.poverty) )
    {
      totalImportance += 3;
      resistanceRaw = resistanceRaw + (float)(*(float *)(disease.poverty) * 3.0);
    }
    if ( *(_DWORD *)(coountry.urban) )
    {
      totalImportance += 2;
      resistanceRaw = resistanceRaw + (float)(*(float *)(disease.urban) + *(float *)(disease.urban));
    }
    if ( *(_DWORD *)(coountry.rural) )
    {
      totalImportance += 2;
      resistanceRaw = resistanceRaw + (float)(*(float *)(disease.rural) + *(float *)(disease.rural));
    }
    if ( *(_DWORD *)(coountry.hot) )
    {
      totalImportance += 10;
      resistanceRaw = resistanceRaw + (float)(*(float *)(disease.hot) * 10.0);
    }
    if ( *(_DWORD *)(coountry.cold) )
    {
      totalImportance += 10;
      resistanceRaw = resistanceRaw + (float)(*(float *)(disease.cold) * 10.0);
    }
    if ( *(_DWORD *)(coountry.humid) )
    {
      totalImportance += 2;
      resistanceRaw = resistanceRaw + (float)(*(float *)(disease.humid) + *(float *)(disease.humid));
    }
    if ( *(_DWORD *)(coountry.arid) )
    {
      resistanceRaw = resistanceRaw + (float)(*(float *)(disease.arid) + *(float *)(disease.arid));
      totalImportance += 2;
    }
  }
LABEL_43:
  if ( apeResistance < 0.2 )
    apeResistance = 0.2;
  *(float *)(localDisease.apeSuitability) = apeResistance;
  if ( totalImportance )
    localResistance = resistanceRaw / (float)totalImportance;
  else
    localResistance = 1.0;
  v27 = *(float *)(a2 + 124) + *(float *)(a3 + 16);
  *(float *)(a1 + 92) = v27;
  if ( *(_DWORD *)(a1 + 200) )
  {
    v28 = *(float *)(a3 + 312);
    if ( *(_DWORD *)(a3 + 12) )
      v28 = v28 + *(float *)(a3 + 316);
    v29 = fmaxf(0.0, v28);
    v27 = (v29 - (float)(v29 * localDisease.infectedPercent)) + v27;
  }
  v30 = v27 * localResistance;
  *(float *)(a1 + 92) = v30;
  if ( v30 < 0.050000001 )
    *(_DWORD *)(a1 + 92) = 1028443341;
  localSeverity = disease.globalSeverity + country.govLocalSeverity;
  localDisease.localSeverity = localSeverity;
  localSeveritySpecial = localSeverity;
  if ( *(_DWORD *)(disease.diseaseType) == 8 && (*(_DWORD *)(a3 + 360) || *(_DWORD *)(a3 + 364)) )
  {
    localSeveritySpecial = localSeverity;
    if ( *(_DWORD *)(coountry.cold) )
    {
      localSeveritySpecial = localSeverity - (float)(localSeverity * 0.2);
      *(float *)(localDisease.localSeverity) = localSeveritySpecial;
    }
    if ( *(_DWORD *)(coountry.arid) )
    {
      localSeveritySpecial = localSeveritySpecial - (float)(localSeveritySpecial * 0.050000001);
      *(float *)(localDisease.localSeverity) = localSeveritySpecial;
    }
    if ( *(_DWORD *)(coountry.poverty) )
    {
      *(float *)(localDisease.localSeverity) = (float)(localSeveritySpecial * 0.050000001) + localSeveritySpecial;
      localSeveritySpecial = (float)(localSeveritySpecial * 0.050000001) + localSeveritySpecial;
    }
    if ( *(_DWORD *)(coountry.wealthy) )
    {
      localSeveritySpecial = localSeveritySpecial - (float)(localSeveritySpecial * 0.050000001);
      *(float *)(localDisease.localSeverity) = localSeveritySpecial;
    }
    if ( *(_DWORD *)(coountry.urban) )
    {
      *(float *)(localDisease.localSeverity) = (float)(localSeveritySpecial * 0.050000001) + localSeveritySpecial;
      localSeveritySpecial = (float)(localSeveritySpecial * 0.050000001) + localSeveritySpecial;
    }
    if ( *(_DWORD *)(coountry.rural) )
    {
      localSeveritySpecial = localSeveritySpecial - (float)(localSeveritySpecial * 0.050000001);
      *(float *)(localDisease.localSeverity) = localSeveritySpecial;
    }
  }
  v33 = *(float *)(a2 + 132) + *(float *)(a3 + 32);
  *(float *)(a1 + 100) = v33;
  *(float *)(a1 + 104) = *(float *)(a3 + 284) + *(float *)(a2 + 128);
  *(float *)(a1 + 516) = *(float *)(a3 + 1300) + *(float *)(a2 + 268);
  localSeverityTrue = localSeveritySpecial;
  *(float *)(a1 + 108) = (float)(int)infectedPercent / (float)(int)*(_QWORD *)(a2 + 24);
  if ( disease.diseaseType <= 8 )
  {
    if ( localSeveritySpecial < 0.05 )
    {
      *(_DWORD *)(localDisease.localSeverity) = 1028443341;
      localSeverityTrue = 0.05;
    }
  }
  else if ( localSeveritySpecial < 0.0 )
  {
    *(_DWORD *)(localDisease.localSeverity) = 0;
    localSeverityTrue = 0.0;
  }
  if ( v33 < 0.0 )
    *(_DWORD *)(a1 + 100) = 0;
  disease.diseaseType = disease.diseaseType;
  environmentalSeverity = localSeverityTrue - localResistance * country.govLocalSeverity;
  if ( disease.diseaseType == 8 )
    environmentalSeverity = fmaxf(disease.globalSeverity, 0.5);
  if ( totalInfected <= 10 )
    SeverityPower = 0.0;
  else
    SeverityPower = environmentalSeverity * environmentalSeverity / 100000.0;
  environmentalSeverityModified = environmentalSeverity;
  if ( disease.diseaseType == 8 )
  {
    environmentalSeverityModifiedPlus = environmentalSeverity * 0.5 + 16.0;
  }
  else
  {
    if ( disease.diseaseType == 7 )
      environmentalSeverityModified = environmentalSeverity * 0.9;
    environmentalSeverityModifiedPlus = environmentalSeverityModified + 30.0;
  }
  severityRate = environmentalSeverity / environmentalSeverityModifiedPlus;
  if ( disease.diseaseType == 8 )
    constK = 1.7;
  else
    constK = 1.5;
  infectedAwareness = localDisease.infectedPercent * localDisease.infectedPercent * constK * severityRate * severityRate + SeverityPower;
  if ( disease.diseaseType != 9 && disease.diseaseType == 8 )
    zombieAwareness = (localDisease.zombiePercent * 5.0) / (localDisease.zombiePercent + 0.2);
  if ( (*(_BYTE *)(a3 + 716) & 2) != 0 )
  {
    localDisease.localPriority = 0;
  }
  else
  {
    deadAwareness = (localDisease.deadPercent * 3.0) / (localDisease.deadPercent + 0.3);
    deadAwarenessModified = qword_180032C58() * deadAwareness;
    dangerAwareness = infectedAwareness + zombieAwareness;
    difficultyConst = 86;
    if ( disease.diseaseType != disease.EDiseaseType.PRION )
      difficultyConst = 96;
    localDisease.localPriority = 
                            fminf(
                              100.0,
                              (((dangerAwareness + deadAwarenessModified) * country.importance)
                                    * (difficultyConst + disease.difficultyVariable))
                            + localDisease.localBonusPriority);
  }
}

public Enum Disease.EDiseaseType
{
	

File: PlagueLibrary/PlagueExternalMP/LocalDiseaseUpdate.c
void __fastcall LocalDiseaseUpdate(double *a1, __int64 a2, __int64 a3)
{
  int v3; 
  __int64 v5; 
  double v8; 
  __int64 v9; 
  double v10; 
  double v11; 
  double v12; 
  double v13; 
  double v14; 
  double v15; 
  double v16; 
  double v17; 
  double v18; 
  double v19; 
  double v20; 
  double v21; 
  double v22; 
  double v23; 
  double v24; 
  double v25; 
  double v26; 
  double v27; 
  double v28; 
  int v29; 
  double v30; 
  int v31; 

  v3 = *(_DWORD *)a1;
  v5 = 0LL;
  *((_DWORD *)a1 + 83) = v3 == disease.superCureCountryNumber;
  *((_DWORD *)a1 + 84) = v3 == disease.nexusCountryNumber;
  if ( disease.uncontrolledAttackBias )
    v8 = 1.0;
  else
    v8 = 0.05000000074505806;
  a1[40] = v8;
  if ( disease._closedBordersSpreadEnhance )
    a1[40] = v8 * 3.0;
  v9 = *((_QWORD *)a1 + 2) + *((_QWORD *)a1 + 4);
  a1[21] = (double)(int)*((_QWORD *)a1 + 1) / (double)(int)disease.globalSeverity;
  v10 = (double)(int)disease.globalSeverity;
  a1[33] = 0.0;
  a1[32] = 0.0;
  a1[28] = 0.0;
  a1[27] = 0.0;
  a1[22] = (double)(int)v9 / v10;
  v11 = 0.0;
  if ( *(_DWORD *)(a2 + 92) )
  {
    v5 = 30LL;
    v11 = disease.wealthy * 30.disease.id.0;
  }
  if ( disease.researchInefficiency )
  {
    v5 = (unsigned int)(v5 + 5);
    v11 = v11 + disease.poverty * 5.0;
  }
  if ( *(_DWORD *)(a2 + 100) )
  {
    v5 = (unsigned int)(v5 + 4);
    v11 = v11 + disease.urban * 4.0;
  }
  if ( disease.globalCureResearch )
  {
    v5 = (unsigned int)(v5 + 4);
    v11 = v11 + disease.rural * 4.0;
  }
  if ( *(_DWORD *)(a2 + 108) )
  {
    v5 = (unsigned int)(v5 + 10);
    v11 = v11 + disease.hot * 10.0;
  }
  if ( disease.globalCureResearchThisTurn )
  {
    v5 = (unsigned int)(v5 + 10);
    v11 = v11 + disease.cold * 10.0;
  }
  if ( *(_DWORD *)(a2 + 116) )
  {
    v5 = (unsigned int)(v5 + 2);
    v11 = v11 + disease.humid * 2.0;
  }
  if ( disease.cureCompletePercent )
  {
    v5 = (unsigned int)(v5 + 2);
    v11 = v11 + disease.arid * 2.0;
  }
  v12 = a1[91];
  if ( v12 != 0.0 && v11 <= v12 )
    v11 = a1[91];
  v13 = a1[92];
  if ( v13 != 0.0 && v13 <= v11 )
    v11 = a1[92];
  if ( (_DWORD)v5 )
    v14 = v11 / (double)(int)v5;
  else
    v14 = 1.0;
  v15 = disease.nexusMinInfect;
  if ( *((_DWORD *)a1 + 84) && v15 - a1[22] * v15 * 3.0 >= 0.0 )
    v16 = v15 - a1[22] * v15 * 3.0;
  else
    v16 = 0.0;
  v17 = (a1[19] + disease.globalInfectiousness + disease.infectedPointsPotAll + v16) * v14;
  a1[14] = v17;
  if ( v17 < 0.1 )
  {
    v5 = 0x3FB999999999999ALL;
    v17 = 0.1;
    a1[14] = 0.1;
  }
  v18 = disease.globalSeverity + disease.globalSeverityPlusLethality;
  a1[15] = v18;
  v19 = disease.globalLethality + disease.dnaPointsGained;
  a1[16] = v19;
  v20 = v19;
  v21 = disease.dnaPointsGained;
  if ( v21 > 0.0 && disease._abilityCostIncrease > 0.0 )
  {
    v20 = v19 - v21;
    a1[16] = v19 - v21;
  }
  a1[17] = disease.corpseTransmission + disease.infectedPointsPot;
  a1[18] = (double)(int)v9 / (double)(int)disease.globalSeverity;
  if ( v18 < 0.0 )
    v18 = 0.0;
  if ( v20 < 0.0 )
    v20 = 0.0;
  v22 = v18 * a1[89] + a1[86];
  v23 = v17 * a1[88] + a1[85];
  v24 = v20 * a1[90] + a1[87];
  a1[15] = v22;
  a1[14] = v23;
  a1[16] = v24;
  v25 = v22 - v14 * disease.globalSeverityPlusLethality;
  if ( v9 <= 10 )
    v26 = 0.0;
  else
    v26 = v25 * v25 / 100000.0;
  v27 = a1[21] * 3.0 / (a1[21] + 0.3);
  v28 = a1[22] * a1[22] * 1.5 * (v25 / (v25 + 30.0)) * (v25 / (v25 + 30.0)) + v26;
  if ( (*(_BYTE *)(a3 + 1060) & 2) != 0 )
  {
    a1[9] = 0.0;
  }
  else
  {
    v29 = disease.diseaseType;
    v30 = qword_180021C90(v5);
    v31 = 86;
    if ( v29 != 4 )
      v31 = 96;
    a1[9] = floorf(fmin(100.0, (v30 * v27 + v28) * disease.cureRequirements * ((double)v31 + disease.difficultyVariable) + a1[10]));
  }
}
